cmd/kube-proxy/app/server.go

```diff
@@ -87,7 +87,8 @@ func (s *ProxyServerConfig) AddFlags(fs *pflag.FlagSet) {
 	fs.IntVar(&s.HealthzPort, "healthz-port", s.HealthzPort, "The port to bind the health check server. Use 0 to disable.")
 	fs.IPVar(&s.HealthzBindAddress, "healthz-bind-address", s.HealthzBindAddress, "The IP address for the health check server to serve on, defaulting to 127.0.0.1 (set to 0.0.0.0 for all interfaces)")
 	fs.IntVar(&s.OOMScoreAdj, "oom-score-adj", s.OOMScoreAdj, "The oom-score-adj value for kube-proxy process. Values must be within the range [-1000, 1000]")
-	fs.StringVar(&s.ResourceContainer, "resource-container", s.ResourceContainer, "Absolute name of the resource-only container to create and run the Kube-proxy in (Default: /kube-proxy).")
+	fs.StringVar(&s.ResourceContainer, "resource-container", "", "[deprecated] this flag doesn't do anything")
+	fs.MarkDeprecated("resource-container", "this flag doesn't do anything")
```

Comment:
Flags are part of the API.  Are we now MANDATING that kube-proxy run in a pod?  I think we should mark it deprecated, but not actually nullify what it does.  We should never have made this be on by default, because now we can not really change the default.  We have to instead pass "" or "/" or something similar from the configs where it runs in a pod.